src/visitors/needsetdata_visitor.hpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2024 Blue Brain Project, EPFL.
+ * See the top-level LICENSE file for details.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+/**
+ * \file
+ * \brief \copybrief nmodl::visitor::NeedSetDataVisitor
+ */
+
+#include <set>
+#include <stack>
+
+#include "ast/block.hpp"
+#include "symtab/decl.hpp"
+#include "visitors/ast_visitor.hpp"
+
+
+namespace nmodl {
+namespace visitor {
+
+/**
+ * \addtogroup visitor_classes
+ * \{
+ */
+
+/**
+ * \class NeedSetDataVisitor
+ * \brief %Visitor for figuring out if function and procedure need setdata call
+ *
+ */
```

Comment:
@iomaganaris : I didn't check all the aspects yet but I "feel" that 1) "visitors" should be something more high level / for more generic purposes 2) "need_set_data" seems quite specific and wondering if we need a visitor here 3) "visitors/*" are increasing and hence these questions

If we just want to know if RANGE or any other variables are being used, can't we use the functionality from existing PerfVisitor where we can get information about the number of reads/writes to variables? That could be also augumented if existing functionality is not sufficient?

Reply:
I see your point regarding the extra visitor. I thought in the beginning that it would be better to create a new visitor to avoid increasing the complexity and adding to the purpose of some other visitor.
Since `need_setdata` is only needed for the code generation probably this fits more in the `CodegenHelperVisitor`.